modular_zubbers/code/modules/storytellers/goals/good/cargo_pod.dm

```diff
@@ -0,0 +1,354 @@
+				possible_cargo += list(/obj/item/stack/sheet/mineral/adamantine = 2)
+			else if(need_category == CARGO_NEED_MORALE)
+				possible_cargo += list(/obj/machinery/vending/boozeomat = 1, /obj/item/instrument/piano_synth = 1)
+				possible_cargo[/obj/machinery/vending/boozeomat] *= base_weight_multiplier
+			else if(need_category == CARGO_NEED_RESEARCH)
+				possible_cargo[/obj/item/research_notes] *= base_weight_multiplier
+				possible_cargo += list(/obj/machinery/rnd/server = 1, /obj/item/stock_parts/servo/pico = 2)
+			amount_of_pods = rand(2, 3)
+
+		if(CARGO_GOOD_LEVEL_EXTREME)  // Extreme: Premium gear + advanced weapons
+			possible_cargo += list(
+				/obj/item/stack/sheet/mineral/adamantine = 3,// Mythical
+				/obj/item/stack/sheet/mineral/runite = 2,          // Rare
+				/obj/item/mod/control/pre_equipped/advanced = 2,   // Elite engi suit
+				/obj/item/gun/energy/lasercannon = 1,              // Powerful laser
+				/obj/item/gun/ballistic/shotgun/bulldog = 1,       // Advanced shotgun
+				/obj/item/storage/belt/security/full = 1,          // Full sec belt
+				/obj/item/beacon = 1							   // Emergency beacon
+			)
+			if(need_category == CARGO_NEED_MEDICAL)
+				possible_cargo += list(/obj/item/storage/medkit/tactical = 3, /obj/item/organ/heart/cybernetic/tier3 = 1)
+				possible_cargo[/obj/item/storage/medkit/tactical] *= base_weight_multiplier
+			else if(need_category == CARGO_NEED_ENGINEERING)
+				possible_cargo[/obj/item/mod/control/pre_equipped/advanced] *= base_weight_multiplier
+				possible_cargo += list(/obj/item/rcd_ammo/large = 2, /obj/item/stack/sheet/mineral/titanium/fifty = 3)
+			else if(need_category == CARGO_NEED_POWER)
+				possible_cargo += list(/obj/item/stock_parts/power_store/cell/bluespace = 3, /obj/machinery/power/rtg/advanced = 1)
+				possible_cargo[/obj/item/stock_parts/power_store/cell/bluespace] *= base_weight_multiplier
+			else if(need_category == CARGO_NEED_SECURITY)
+				possible_cargo[/obj/item/gun/energy/lasercannon] *= base_weight_multiplier
+				possible_cargo += list(/obj/item/gun/ballistic/rocketlauncher = 1, /obj/item/clothing/suit/armor/hos = 1)
+			else if(need_category == CARGO_NEED_RESOURCES)
+				possible_cargo += list(/obj/item/stack/sheet/mineral/bananium = 2, /obj/item/stack/sheet/mineral/abductor = 1)
+				possible_cargo[/obj/item/stack/sheet/mineral/adamantine] *= base_weight_multiplier
+			else if(need_category == CARGO_NEED_MORALE)
+				possible_cargo += list(/obj/item/storage/box/donkpockets = 3, /obj/machinery/computer/arcade = 1)  // Food and games
+				possible_cargo[/obj/item/storage/box/donkpockets] *= base_weight_multiplier
+			else if(need_category == CARGO_NEED_RESEARCH)
+				possible_cargo += list(/obj/item/disk/design_disk/long_range_pda = 2, /obj/item/stock_parts/scanning_module/phasic = 1)
+				possible_cargo[/obj/item/disk/design_disk/long_range_pda] *= base_weight_multiplier
+			amount_of_pods = 3
```

Comment:
why not just have a assoc list pregenerated with valid paths based on CARGO_GOOD_LEVEL_EXTREME, or several of them?
this is kinda ungood in general